app/models/person/add_request.rb

```diff
@@ -66,4 +68,24 @@ def requester_full_roles
     end
   end
 
+  private
+
+  def last_layer_group
+    last_role = Role.unscoped.where(person: person).order(:deleted_at).last
+    last_role && last_role.group.layer_group
+  end
+
+  def self.people_with_last_group(group)
+    candidate_ids = no_role_and_last_member_in(group).map(&:id)
+    Person::AddRequest.joins(:person).where(people: { id: candidate_ids })
+  end
+
+  def self.no_role_and_last_member_in(group)
+    subquery = "select max(roles.deleted_at) from roles " \
+               "where roles.person_id = people.id and people.primary_group_id is null"
+    query =    "select people.* from people inner join roles on roles.person_id = people.id " \
+               "inner join groups on roles.group_id = groups.id " \
+               "where roles.deleted_at = (#{subquery}) and groups.layer_group_id = ?;"
+    Person.find_by_sql([query, group.id])
+  end
```

Comment:
Klassenmethoden oben in den `class << self` Block gruppieren.
Gleiche Issues wie bei [129](https://github.com/hitobito/hitobito/pull/129/files/90d35111222ff6427ad3351e27d9d790a6e6fc4e#diff-95e71e63c60916e0e9e95a67bb5cc4a2) 
Möglichst auch gleich diesen Code dort verwenden und nicht duplizieren.